lib/wallaroo/core/source/tcp_source/tcp_source.pony

```diff
@@ -333,8 +334,17 @@ actor TCPSource is Producer
     source.
     """
 
+    @printf[I32]("Shutting down TCPSource\n".cstring())
     _hard_close()
-
+    _router_registry.unregister_source(this)
```

Comment:
If the outgoing boundaries created for a TCPSource are registered as disposables with Connections, I don't think you're disposing them here when the TCPSource disposes